delivery_purchase/models/purchase_order.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2015 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# Copyright 2016 Tecnativa - Pedro M. Baeza
+# Copyright 2023 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.tools.float_utils import float_is_zero
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    carrier_id = fields.Many2one(
+        comodel_name="delivery.carrier",
+        string="Delivery Method",
+    )
+    delivery_price = fields.Float(
+        compute="_compute_delivery_price", store=True, readonly=False
+    )
+
+    @api.onchange("partner_id")
+    def onchange_partner_id_delivery_purchase(self):
```

Comment:
I would have used `carrier_id` declared as computed/stored/readonly=False in order to use compute here and not onchange.